src/components/Pressable.tsx

```diff
@@ -0,0 +1,277 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { GestureObjects as Gesture } from '../handlers/gestures/gestureObjects';
+import { GestureDetector } from '../handlers/gestures/GestureDetector';
+import {
+  GestureStateChangeEvent,
+  GestureTouchEvent,
+  TouchData,
+} from '../handlers/gestureHandlerCommon';
+import { PressEvent, PressableEvent, PressableProps } from './PressableProps';
+import { HoverGestureHandlerEventPayload } from '../handlers/gestures/hoverGesture';
+import { LongPressGestureHandlerEventPayload } from '../handlers/LongPressGestureHandler';
+import { Insets, StyleProp, View, ViewStyle } from 'react-native';
+import RNButton from './GestureHandlerButton';
+
+const DEFAULT_LONG_PRESS_DURATION = 500;
+const DEFAULT_HOVER_DELAY = 0;
+
+const numberAsInset = (value: number): Insets => ({
+  left: value,
+  right: value,
+  top: value,
+  bottom: value,
+});
+
+const addInsets = (a: Insets, b: Insets): Insets => ({
+  left: (a.left ?? 0) + (b.left ?? 0),
+  right: (a.right ?? 0) + (b.right ?? 0),
+  top: (a.top ?? 0) + (b.top ?? 0),
+  bottom: (a.bottom ?? 0) + (b.bottom ?? 0),
+});
+
+const touchToPressEvent = (data: TouchData, timestamp: number): PressEvent => ({
+  identifier: data.id,
+  locationX: data.x,
+  locationY: data.y,
+  pageX: data.absoluteX,
+  pageY: data.absoluteY,
+  target: 0, // fixme if possible, set to correct target ID
+  timestamp: timestamp,
+  touches: [], // intentionally empty
+  changedTouches: [], // intentionally empty
+});
+
+const changeToTouchData = (
+  event: GestureStateChangeEvent<
+    HoverGestureHandlerEventPayload | LongPressGestureHandlerEventPayload
+  >
+): TouchData => ({
+  id: 0, // fixme if possible, set to correct pointer ID
+  x: event.x,
+  y: event.y,
+  absoluteX: event.absoluteX,
+  absoluteY: event.absoluteY,
+});
+
+const isTouchWithinInset = (
+  // all of these are guaranteed to be their type,
+  // but we cannot use '!' anymore
```

Comment:
What do you mean?

Reply:
My bad, I though Eslint prohibited usage of `!`, but it doesn't seem like a problem here.

Reply:
![image](https://github.com/software-mansion/react-native-gesture-handler/assets/74246391/6c3a8f04-bd0f-42d2-bbd0-f2d6659bada1)

Reply:
Why mark them as nullable (`?`) though? It's called in one place and 2 of these 3 arguments seem to be always defined.

Reply:
Because the Inset type explicitly has every of it's child members as `number | undefined`, and so I cannot just use them as it is. 

Same goes for the event types, which while they are certain to have a `number` type, the events themselves are `event | undefined`.

And for the `touch?: TouchData` object, it's just a choice between having it non-null checked either here in `utils`, or in `Pressable`, where the passed argument is read from an array, which implicitly makes it `| undefined`

Reply:
> The Inset type explicitly has every of it's child members as number | undefined, and so I cannot just use them as it is.

But `inset` itself seems to always be defined (it's always `normalizedHitslop`), no?

> And for the touch?: TouchData object, it's just a choice between having it non-null checked either here in utils, or in Pressable, where the passed argument is read from an array, which implicitly makes it | undefined.

You're right.

Reply:
> But inset itself seems to always be defined (it's always normalizedHitslop), no?

You're right, I'm correcting this mistake.